server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -258,6 +258,13 @@ public Analyzer getSearchAnalyzer(MappedFieldType fieldType) {
         return mapperService.searchAnalyzer();
     }
 
+    /**
+     * Returns the default search analyzer
+     */
+    public Analyzer getSearchAnalyzer() {
+        return this.mapperService.searchAnalyzer();
+    }
```

Comment:
possibly, searchQuoteAnalyzer and searchAnalyzer could be removed from MapperService as they are only used through QueryShardContext. Only they are now a subclass of `MapperAnalyzerWrapper` which is package protected within MapperService and used also for indexAnalyzer which must stay where it is.

Reply:
Yes, MapperService builds the per-field analyzer here so we need to keep it, I think.  Although we may be able to remove the per-field delegation for search/searchQuote because generally speaking queries get their analyzers directly from the field types.  We can look at it later.